### DISCUSSION TITLE
SOLVED → What would be a good way of supporting spaces in paths?

### DISCUSSION THREAD

**Discussion by Gaweringo** [Zeioth/compiler.nvim#38] [General]
I just had the issue, that I was not able to use `Run Makefile` in directory that has a space in the path. After looking around in the plugin code, I narrowed it down to the fact that for the overseer task, the command gets the file path just inserted into the string:
https://github.com/Zeioth/compiler.nvim/blob/a0fc34e3aa6c8109863441fc674cf04de6241bd0/lua/compiler/languages/make.lua#L27-L36
So that a file path with a space in it ends up being run like this: `make -f C:\My\File Path With\Space\Makefile` which does not work.
I was wondering if this is something that should be looked at and changed.

Maybe by formatting the file path before concatenating it into the cmd string with `string.format("%q", makefile)` which would escape special characters and add quotes around the string. Would turn the path from before into:
`make -f "C:\\My\\File Path With\\Space\\Makefile"`

If this is something that others would also like to see change, I would be happy to open an issue or make a PR, although I am unsure if wrapping every path that gets put into the `cmd` of a task is the best solution.

What do you think?

**Comment by Zeioth** [comment]
@Gaweringo this has been fixed in https://github.com/Zeioth/compiler.nvim/releases/tag/v3.3.5
